compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -884,6 +884,7 @@ public void visit(BLangUnionTypeNode unionTypeNode) {
                         memBType.tag == TypeTags.UNION && !Symbols.isFlagOn(memBType.tsymbol.flags, Flags.TYPE_PARAM) ?
                                 ((BUnionType) memBType).getMemberTypes().stream() :
                                 Stream.of(memBType))
+                .filter(memBType -> memBType.tag != TypeTags.NEVER)
```

Comment:
Instead of handling this here, I think it'd be better to add this logic to the API for creating unions in `BUnionType` itself. That way, this behaviour will be guaranteed for any union created, not just the ones created through the resolver.

Reply:
Ack